libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyFile.java

```diff
@@ -0,0 +1,311 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.secure_sm.policy;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.security.CodeSource;
+import java.security.Permission;
+import java.security.PermissionCollection;
+import java.security.Permissions;
+import java.security.ProtectionDomain;
+import java.security.cert.Certificate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@SuppressWarnings("removal")
+public class PolicyFile extends java.security.Policy {
+
+    private volatile PolicyInfo policyInfo;
+    private URL url;
+
+    /**
+     * When a policy file has a syntax error, the exception code may generate
+     * another permission check and this can cause the policy file to be parsed
+     * repeatedly, leading to a StackOverflowError or ClassCircularityError.
+     * To avoid this, this set is populated with policy files that have been
+     * previously parsed and have syntax errors, so that they can be
+     * subsequently ignored.
+     */
+    private static Set<URL> badPolicyURLs = Collections.newSetFromMap(new ConcurrentHashMap<URL, Boolean>());
+
+    /**
+     * Initializes the Policy object and reads the default policy
+     * from the specified URL only.
+     */
+    public PolicyFile(URL url) {
+        this.url = url;
+        init(url);
+    }
+
+    /**
+     * Initializes the Policy object and reads the default policy
+     * configuration file(s) into the Policy object.
+     *
+     * See the class description for details on the algorithm used to
+     * initialize the Policy object.
+     */
+    private void init(URL url) {
+        PolicyInfo newInfo = new PolicyInfo();
+        initPolicyInfof(url, newInfo);
+        policyInfo = newInfo;
+    }
+
+    private void initPolicyInfof(URL policy, PolicyInfo newInfo) {
```

Comment:
```suggestion
    private void initPolicyInfo(URL policy, PolicyInfo newInfo) {
```

Reply:
i think there was a typo. will fix it out.